### PR TITLE
Add Allergies exercise

### PR DESCRIPTION
**Pull request by robtimp**

### CODE REVIEW

**Comment by robtimp** [comment]
@chillpop @masters3d Got another one for ya.

**Comment by chillpop** [comment]
This looks good.  Can you add another test at the beginning that only tests for one allergy like [so](https://github.com/exercism/xjava/blob/master/exercises/allergies/src/test/java/AllergiesTest.java#L21-L25)?

File: exercises/allergies/AllergiesTest.m [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#import <XCTest/XCTest.h>
+
+#if __has_include("AllergiesExample.h")
```
Comment (chillpop) [inline exercises/allergies/AllergiesTest.m]: FYI, I think you missed this in your previous PR. (I missed it too)

**Comment by chillpop** [comment]
LGTM. Thanks again Rob!

File: exercises/allergies/AllergiesTest.m (hunk 1 above)
Comment (robtimp) [inline exercises/allergies/AllergiesTest.m]: The 'All Your Base' exercise? It has it, too.

Comment (chillpop) [inline exercises/allergies/AllergiesTest.m]: You're right. I was looking the wrong place. 😕

**Pull request merged**